src/components/Posts/ComposePost.tsx

```diff
@@ -24,21 +31,28 @@ const styles = {
   postButton: { borderRadius: 5 },
 };
 
-const ComposePost = () => {
+const ComposePost = ({ placeholder }: ComposePostProps) => {
   const [postTextContent, setPostTextContent] = useState("");
   const user = useAppSelector((state) => state.user);
+  const dispatch = useAppDispatch();
 
   const onSubmit = async (e: React.SyntheticEvent) => {
     e.preventDefault();
     try {
       const textContent = postTextContent;
-      await axios.post("http://localhost:3001/api/posts", {
+      const newPost = await axios.post("http://localhost:3001/api/posts", {
         userId: user.userId,
         textContent,
       });
       setPostTextContent("");
-      // TODO: Update posts in postContext so that the page rerenders rather than refreshing
-      window.location.reload();
+      dispatch(
+        appendPost({
+          ...newPost.data,
+          username: user.username,
+          displayName: user.displayName,
+          numberOfLikes: 0,
```

Comment:
I agree with this solution for the NaN thing, but can we please move this into the reducer? It'll save us from having to set the number of likes in every dispatch